gc/gc.go

```diff
@@ -172,12 +172,16 @@ func ColoredSet(ctx context.Context, pn pin.Pinner, ng ipld.NodeGetter, bestEffo
 	// disk backed to conserve memory.
 	errors := false
 	gcs := cid.NewSet()
-	getLinks := func(ctx context.Context, cid cid.Cid) ([]*ipld.Link, error) {
-		links, err := ipld.GetLinks(ctx, ng, cid)
+	getLinks := func(ctx context.Context, link cid.Cid) ([]*ipld.Link, error) {
+		if link.Type() == cid.Raw {
+			return nil, nil
+		}
+
+		links, err := ipld.GetLinks(ctx, ng, link)
```

Comment:
`GetLinks` seems to already have fast path for raw blocks.

```go
func GetLinks(ctx context.Context, ng NodeGetter, c cid.Cid) ([]*Link, error) {
	if c.Type() == cid.Raw {
		return nil, nil
	}
	...
```

Reply:
Well... Now really I feel like an idiot.

@balajijinnah I'm so sorry. Apparently, we already _have_ this optimization, I just never thought to actually check.

Reply:
Lol, Even I didn't even check it. 

It's a good experience to contribute. Looking to contribute more.